values/encoding/string_encode.go

```diff
@@ -112,12 +111,25 @@ func (enc *stringEncoder) Encode(strVals values.StringValues, writer io.Writer)
 
 	switch metaProto.Encoding {
 	case encodingpb.EncodingType_RAW_SIZE:
-		return enc.rawSizeEncode(valuesIt, writer)
+		if err := enc.rawSizeEncode(valuesIt, writer); err != nil {
+			return err
+		}
 	case encodingpb.EncodingType_DICTIONARY:
-		return enc.dictionaryEncode(valuesIt, dictionary, writer)
+		if err := enc.dictionaryEncode(valuesIt, dictionary, writer); err != nil {
+			return err
+		}
 	default:
 		return fmt.Errorf("invalid encoding type: %v", metaProto.Encoding)
 	}
+
+	// Close the compressWriter if its present.
+	if compressWriter != nil {
```

Comment:
Any reason not to `defer compressWriter.Close` at line 100?

Reply:
Because we wanted to surface the error that it returns to the caller.

Reply:
Because we wanted to surface the error that the writer returns to the caller.